runtime/src/iree-amd-aie/driver/xrt-lite/registration/driver_module.c

```diff
@@ -0,0 +1,144 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/driver/xrt-lite/registration/driver_module.h"
+
+#include "iree-amd-aie/driver/xrt-lite/api.h"
+#include "iree/base/api.h"
+#include "iree/base/internal/flags.h"
+
+IREE_FLAG(int32_t, xrt_lite_n_core_rows, 4,
+          "Number of core rows to use on NPU.");
+IREE_FLAG(int32_t, xrt_lite_n_core_cols, 4,
+          "Number of core cols to use on NPU.");
```

Comment:
`4` is the default here? Shouldn't we just fail if not provided, as this might create issues in the future?

Reply:
Is there a config where 4x4 isn't correct? I would rather have a poor, but not failing, experience in the case someone forgets the flag (it's quite annoying to me that these flags are globals/CLI rather than somehow passed/tunnelled through the IR).

Reply:
Maybe you're right tho

Reply:
> it's quite annoying to me that these flags are globals/CLI rather than somehow passed/tunnelled through the IR

Yeah, we should be able to do that in the future.

Reply:
Okay I've made them required and added checks/fails if they're not provided.